### PR TITLE
SLR - SL-114 - Error messages

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: [SL-114](https://stellarwp.atlassian.net/browse/SL-114)

This updates the code to add a Service status check and change the UI behaviour depending on said status.

[Demo](https://share.cleanshot.com/Pb6jTlhQ)

[SL-114]: https://stellarwp.atlassian.net/browse/SL-114?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Tickets/Seating/app/blockEditor/capacity-form/service-error.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+				<span style={style}>
+					{__(
+						'Your site is not connected to the Seat Builder service.',
+						'event-tickets'
+					)}{' '}
+					<a
+						style={anchorStyle}
+						href={serviceConnectUrl}
+						target="_blank"
+						rel="noreferrer"
```
Comment (rafsuntaskin) [inline src/Tickets/Seating/app/blockEditor/capacity-form/service-error.js:37]:
```suggestion
						rel="noreferrer noopener"
```

File: src/Tickets/Seating/app/blockEditor/capacity-form/service-error.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+				<span style={style}>
+					{__(
+						'Your license for Seating has expired.',
+						'event-tickets'
+					)}{' '}
+					<a
+						style={anchorStyle}
+						href="https://evnt.is/1bdu"
+						target="_blank"
+						rel="noreferrer"
```
Comment (rafsuntaskin) [inline src/Tickets/Seating/app/blockEditor/capacity-form/service-error.js:58]:
```suggestion
						rel="noreferrer noopener"
```

File: src/Tickets/Seating/app/blockEditor/filters.js [hunk 3]
```diff
@@ -145,6 +148,21 @@ addAction(
...
+addFilter(
+	'tec.tickets.blocks.Ticket.actionItems',
+	'tec.tickets.seating',
+	removeAllActionsFromTicket
+);
+
+addFilter(
+	'tec.tickets.blocks.Ticket.isSelected',
+	'tec.tickets.seating',
+	disableTicketSelection
```
Comment (rafsuntaskin) [inline src/Tickets/Seating/app/blockEditor/filters.js:165]:
```suggestion
	disableTicketSelection,
```

File: src/Tickets/Seating/app/blockEditor/filters.js [hunk 4]
```diff
@@ -11,6 +11,9 @@ import {
 	filterTicketIsAsc,
 	setSeatTypeForTicket,
 	filterSettingsFields,
+	disableConfirmInTicketDashboard,
+	removeAllActionsFromTicket,
+	disableTicketSelection
```
Comment (rafsuntaskin) [inline src/Tickets/Seating/app/blockEditor/filters.js:16]:
```suggestion
	disableTicketSelection,
```

File: src/Tickets/Seating/app/blockEditor/filters.js (hunk 3 above)
Comment (rafsuntaskin) [inline src/Tickets/Seating/app/blockEditor/filters.js:165]: This filter looks broken, bad merge maybe.

File: src/views/seating/tickets-block-error.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
+<?php
+/**
+ * Seating tickets block error template.
+ *
+ * Override this template in your own theme by creating a file at
+ * [your-theme]/tribe/tickets-seating/tickets-block-error.php
+ *
+ * @since   TBD
```
Comment (rafsuntaskin) [inline src/views/seating/tickets-block-error.php:8]:
```suggestion
 * @since TBD
```

File: src/Tickets/Seating/Service/Error_Content.php [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+					'You need to connect your site to the Seat Builder in order to create Seating Maps and Seat Layouts.',
+					'event-tickets'
+				);
+				$cta_label = _x( 'Connect', 'Connect to the Seat Builder button label', 'event-tickets' );
+				$cta_url   = admin_url( 'admin.php?page=tec-tickets-settings&tab=licenses' );
+				break;
+			case Service_Status::INVALID_LICENSE:
+				$renew_link = sprintf(
+					// translators: %s is the renew link label.
+					'<a href="https://evnt.is/1bdu" target="_blank">%s</a>',
```
Comment (rafsuntaskin) [inline src/Tickets/Seating/Service/Error_Content.php:81]:
```suggestion
					'<a href="https://evnt.is/1bdu" target="_blank" rel="noreferrer noopener">%s</a>',
```

File: src/Tickets/Seating/Service/Error_Content.php [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+						'Your site is not connected to the Seat Builder service. You need to connect your site to use assigned seating. %s',
+						'event-tickets'
+					),
+					$connect_link
+				);
+				break;
+			case Service_Status::INVALID_LICENSE:
+				$renew_link = sprintf(
+					// translators: %s is the renew link label.
+					'<a href="https://evnt.is/1bdu" target="_blank">%s</a>',
```
Comment (rafsuntaskin) [inline src/Tickets/Seating/Service/Error_Content.php:146]:
```suggestion
					'<a href="https://evnt.is/1bdu" target="_blank" rel="noreferrer noopener">%s</a>',
```

**Review by rafsuntaskin — CHANGES_REQUESTED** [review]

File: src/Tickets/Seating/Service/Error_Content.php [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+				break;
+		}
+
+		return wp_kses(
+			$message,
+			[
+				'a' => [
+					'href'   => [],
+					'target' => [],
+				],
```
Comment (dpanta94) [inline src/Tickets/Seating/Service/Error_Content.php:166]:
```suggestion
				'a' => [
					'href'   => [],
					'target' => [],
					'rel'    => [],
				],
```

File: tests/slr_jest/blockEditor/hook-callbacks.spec.js [hunk 9]
```diff
@@ -29,26 +31,26 @@ describe('hook-callbacks', () => {
...
 			);
 		});
 	});
 
-	describe('filterSetBodyDetails', () => {
-		it('should modify body', () => {
-			const seatType = 'seat-type-uuid-1';
-			const layoutId = 'layout-uuid-1';
-			const eventCapacity = '40';
+	describe("filterSetBodyDetails", () => {
```
Comment (dpanta94) [inline tests/slr_jest/blockEditor/hook-callbacks.spec.js:49]: is these changes your IDE's changes ? Mine doesn't catch those with the linter.

Comment (lucatume) [reply tests/slr_jest/blockEditor/hook-callbacks.spec.js:49]: If you're talking about style and formatting changes, I'm using prettier and jslint.

**Review by dpanta94 — APPROVED** [review]

**Pull request merged**